src/services/bunker_cases/midterm_slashing_penalty.py

```diff
@@ -24,56 +23,50 @@ class MidtermSlashingPenalty:
     def is_high_midterm_slashing_penalty(
         blockstamp: ReferenceBlockStamp,
         frame_config: FrameConfig,
-        all_validators: dict[str, Validator],
-        lido_validators: dict[str, LidoValidator],
-        cl_rebase: Gwei
+        all_validators: list[Validator],
+        lido_validators: list[LidoValidator],
+        frame_cl_rebase: Gwei
     ) -> bool:
         logger.info({"msg": "Detecting high midterm slashing penalty"})
-        all_slashed_validators = MidtermSlashingPenalty.not_withdrawn_slashed_validators(
+        all_slashed_validators = MidtermSlashingPenalty.get_not_withdrawn_slashed_validators(
             all_validators, blockstamp.ref_epoch
         )
-        lido_slashed_validators = MidtermSlashingPenalty.not_withdrawn_slashed_validators(
+        lido_slashed_validators = MidtermSlashingPenalty.get_not_withdrawn_slashed_validators(
             lido_validators, blockstamp.ref_epoch
         )
         logger.info({"msg": f"Slashed: All={len(all_slashed_validators)} | Lido={len(lido_slashed_validators)}"})
 
-        future_midterm_penalty_lido_slashed_validators = {
-            k: v for k, v in lido_slashed_validators.items()
-            if MidtermSlashingPenalty.get_midterm_slashing_epoch(v) > blockstamp.ref_epoch
-        }
+        # Put all Lido slashed validators to future frames by midterm penalty epoch
+        future_frames_lido_validators = MidtermSlashingPenalty.get_per_frame_lido_validators_with_future_midterm_epoch(
+            blockstamp.ref_epoch, frame_config, lido_slashed_validators
+        )
+
         # If no one Lido in current not withdrawn slashed validators
         # and no one midterm slashing epoch in the future - no need to bunker
-        if not future_midterm_penalty_lido_slashed_validators:
+        if not future_frames_lido_validators:
             return False
 
-        # We should calculate total_balance for each bucket, but we do it once for all per_epoch_buckets
-        total_balance = calculate_total_active_effective_balance(list(all_validators.values()), blockstamp.ref_epoch)
-        # Calculate lido midterm penalties in each epoch where lido slashed
-        per_epoch_buckets = MidtermSlashingPenalty.get_per_epoch_buckets(all_slashed_validators, blockstamp.ref_epoch)
-        per_epoch_lido_midterm_penalties = MidtermSlashingPenalty.get_per_epoch_lido_midterm_penalties(
-            per_epoch_buckets, future_midterm_penalty_lido_slashed_validators, total_balance
-        )
-        # Calculate lido midterm penalties impact in each frame
-        per_frame_buckets = MidtermSlashingPenalty.get_per_frame_lido_midterm_penalties(
-            per_epoch_lido_midterm_penalties, frame_config
-        )
+        # We should calculate total balance for each midterm penalty epoch, but we can't do it for the future epochs
```

Comment:
> We should calculate, ... but we can't 

Let's just say that we make projection based on the current state of the chain

Reply:
fixed [here](https://github.com/lidofinance/lido-oracle/pull/261/commits/a8f2b0963e05fff55623cb32bc398da9e50c3458)